javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java

```diff
@@ -302,6 +304,16 @@ public static ClientModel getClientModel(String name) {
         return getClientModelFunction.apply(name);
     }
 
+    public static boolean isClientModel(IType type) {
+        if (type instanceof ClassType) {
+            ClassType classType = (ClassType) type;
+            return classType.getPackage().startsWith(JavaSettings.getInstance().getPackage())
+                    && getClientModel(classType.getName()) != null;
+        } else {
+            return false;
+        }
+    }
```

Comment:
Use this to make a more solid inference on whether the `type` is a client model,
than previous
```java
type instanceof ClassType && ((ClassType) type).getPackage().startsWith(JavaSettings.getInstance().getPackage())
```